packages/ui/src/components/Chip/ChipIcon.tsx

```diff
@@ -0,0 +1,84 @@
+import styled from '@emotion/styled'
+import * as Icon from '@ultraviolet/icons'
+import { useContext } from 'react'
+import type { PascalToCamelCaseWithoutSuffix } from 'src/types'
+import { ChipContext } from './ChipContext'
+
+const Container = styled.button`
+  background: none;
+  border:none;
+  cursor: pointer;
+  padding:0;
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  &[data-type="trailing"]{
+    margin-right: -${({ theme }) => theme.space['0.5']};
+  }
+
+  &[data-has-onclick="true"][data-active="false"]:hover{
+      background-color: ${({ theme }) => theme.colors.neutral.backgroundStrongHover};
+  }
+
+  &[data-has-onclick="true"][data-active="true"]:hover{
+      background-color: ${({ theme }) => theme.colors.primary.backgroundStrong};
+  }
+
+  &[data-disabled="true"]{
+    cursor: not-allowed;
+  }
+`
+
+type IconType = PascalToCamelCaseWithoutSuffix<keyof typeof Icon, 'Icon'>
+
+type ChipIconType = {
+  /**
+   * Add an icon in the chip
+   */
+  name: IconType
+  type: 'trailing' | 'leading'
```

Comment:
Seems like you use this to specify if the icon is on the left or right of the content. I think you can remove this and manage the spacing in the Chip parent directly using stack and gap. wdyt?